src/analysis/calculus/uniform_limits_deriv.lean

```diff
@@ -0,0 +1,567 @@
+end
+
+/-- `(d/dx) lim_{n → ∞} f n x = lim_{n → ∞} f' n x` when the `f' n` converge
+_uniformly_ to their limit on an open set containing `x`. -/
+lemma has_fderiv_at_of_tendsto_uniformly_on
+  {s : set E} (hs : is_open s)
+  (hf : ∀ (n : ι), ∀ (x : E), x ∈ s → has_fderiv_at (f n) (f' n x) x)
+  (hfg : ∀ (x : E), x ∈ s → tendsto (λ n, f n x) l (𝓝 (g x)))
+  (hfg' : tendsto_uniformly_on f' g' l s) :
+  ∀ (x : E), x ∈ s → has_fderiv_at g (g' x) x :=
+begin
+  intros x hx,
+  have hf : ∀ᶠ (n : ι × E) in (l ×ᶠ 𝓝 x), has_fderiv_at (f n.fst) (f' n.fst n.snd) n.snd,
+  { exact eventually_prod_iff.mpr ⟨(λ y, true), (by simp), (λ y, y ∈ s),
+      eventually_mem_set.mpr (mem_nhds_iff.mpr ⟨s, rfl.subset, hs, hx⟩),
+      (λ n hn y hy, hf n y hy)⟩, },
+
+  have hfg : ∀ᶠ y in 𝓝 x, tendsto (λ n, f n y) l (𝓝 (g y)),
+  { exact eventually_iff.mpr (mem_nhds_iff.mpr ⟨s, set.subset_def.mpr hfg, hs, hx⟩), },
+
+  have hfg' := hfg'.tendsto_uniformly_on_filter.mono_right (calc
+    𝓝 x = 𝓝[s] x : ((hs.nhds_within_eq hx).symm)
+    ... ≤ 𝓟 s : (by simp only [nhds_within, inf_le_right])),
+
+  exact has_fderiv_at_of_tendsto_uniformly_on_filter hf hfg hfg',
+end
+
+/-- `(d/dx) lim_{n → ∞} f n x = lim_{n → ∞} f' n x` when the `f' n` converge
+_uniformly_ to their limit. -/
+lemma has_fderiv_at_of_tendsto_uniformly
+  (hf : ∀ (n : ι), ∀ (x : E), has_fderiv_at (f n) (f' n x) x)
+  (hfg : ∀ (x : E), tendsto (λ n, f n x) l (𝓝 (g x)))
+  (hfg' : tendsto_uniformly f' g' l) :
+  ∀ (x : E), has_fderiv_at g (g' x) x :=
+begin
+  intros x,
+  have hf : ∀ (n : ι), ∀ (x : E), x ∈ set.univ → has_fderiv_at (f n) (f' n x) x, { simp [hf], },
+  have hfg : ∀ (x : E), x ∈ set.univ → tendsto (λ n, f n x) l (𝓝 (g x)), { simp [hfg], },
+  have hfg' : tendsto_uniformly_on f' g' l set.univ, { rwa tendsto_uniformly_on_univ, },
+  refine has_fderiv_at_of_tendsto_uniformly_on is_open_univ hf hfg hfg' x (set.mem_univ x),
+end
+
+end limits_of_derivatives
+
+section deriv
+
+/-! ### `deriv` versions of above theorems -/
+
+variables {ι : Type*} {l : filter ι} [ne_bot l]
+  {𝕜 : Type*} [is_R_or_C 𝕜]
+  {G : Type*} [normed_add_comm_group G] [normed_space 𝕜 G]
+  {f : ι → 𝕜 → G} {g : 𝕜 → G} {f' : ι → 𝕜 → G} {g' : 𝕜 → G}
+  {x : 𝕜}
+
+lemma uniform_cauchy_seq_on_filter_of_tendsto_uniformly_on_filter_deriv
+  (hf : ∀ᶠ (n : ι × 𝕜) in (l ×ᶠ 𝓝 x), has_deriv_at (f n.fst) (f' n.fst n.snd) n.snd)
+  (hfg : tendsto (λ n, f n x) l (𝓝 (g x)))
+  (hfg' : uniform_cauchy_seq_on_filter f' l (𝓝 x)) :
+  uniform_cauchy_seq_on_filter f l (𝓝 x) :=
+begin
+  -- The first part of the proof rewrites `hf` and the goal to be functions so that Lean
+  -- can recognize them when we apply
+  -- `uniform_cauchy_seq_on_filter_of_tendsto_uniformly_on_filter_fderiv`
+  let F' : ι → 𝕜 → (𝕜 →L[𝕜] G) := (λ n, λ z, (1 : 𝕜 →L[𝕜] 𝕜).smul_right (f' n z)),
+  simp_rw has_deriv_at_iff_has_fderiv_at at hf,
+  have : ∀ n : ι, ∀ z : 𝕜, (1 : 𝕜 →L[𝕜] 𝕜).smul_right (f' n z) = F' n z, simp,
+  simp_rw this at hf,
```

Comment:
These two lines become unnecessary after reordering the arguments of `uniform_cauchy_seq_on_filter_of_tendsto_uniformly_on_filter_fderiv` (probably same for the proofs below)

Reply:
Done and yes it did figure it out now!